src/main/scala/org/constellation/snapshotstreaming/opensearch/UpdateRequestBuilder.scala

```diff
@@ -10,89 +10,117 @@ import com.sksamuel.elastic4s.requests.update.UpdateRequest
 import org.constellation.snapshotstreaming.SnapshotProcessor.GlobalSnapshotWithState
 import org.constellation.snapshotstreaming.opensearch.mapper.CurrencySnapshotMapper
 import org.constellation.snapshotstreaming.opensearch.mapper.GlobalSnapshotMapper
-import org.constellation.snapshotstreaming.opensearch.schema._
 import org.constellation.snapshotstreaming.Configuration
+import org.constellation.snapshotstreaming.opensearch.SplitSnapshot.splitAndHash
+import org.constellation.snapshotstreaming.schema.{AddressBalance, Block, CurrencyData, CurrencySnapshot, FeeTransaction, Snapshot, Transaction}
 
 case class UpdateRequests(
   sequentialRequests: Seq[Seq[UpdateRequest]],
   parallelRequests  : List[List[UpdateRequest]]
 )
 
+case class SplitSnapshot(snapshot: Snapshot, blocks: Seq[Block], txs: List[Transaction], balances: Seq[AddressBalance], cdSnapshots: Seq[CurrencyData[Snapshot]], cdCcySnapshots: Seq[CurrencyData[CurrencySnapshot]], cdBlocks: Seq[CurrencyData[Block]], cdTxs: Seq[CurrencyData[Transaction]], cdFeeTxs: Seq[CurrencyData[FeeTransaction]], cdBalances: Seq[CurrencyData[AddressBalance]])
+object SplitSnapshot {
+  def splitAndHash[F[_]: Async](
+                                 globalMapper: GlobalSnapshotMapper[F],
+                                 currencyMapper: CurrencySnapshotMapper[F],
+                                 hasher: Hasher[F],
+                                 txHasher: Hasher[F]
+                               )(globalSnapshotWithState: GlobalSnapshotWithState,
+                                 timestamp: Date
+                               ): F[SplitSnapshot] = {
+
+
+
+    val GlobalSnapshotWithState(globalSnapshot, maybePrevSnapshotInfo, snapshotInfo, currencySnapshots) =
+      globalSnapshotWithState
+    for {
+      mappedGlobalData <- globalMapper.mapGlobalSnapshot(
+        globalSnapshot,
+        maybePrevSnapshotInfo,
+        snapshotInfo,
+        timestamp,
+        txHasher,
+        hasher
+      )
+      mappedCurrencyData <- currencyMapper.mapCurrencySnapshots(
+        currencySnapshots,
+        maybePrevSnapshotInfo.map(_.lastCurrencySnapshots),
+        timestamp,
+        txHasher,
+        hasher
+      )
+      (currSnapshot, currIncrementalSnapshots, currBlocks, currTransactions, currFeeTransactions, currBalances) =
+        mappedCurrencyData
+    } yield {
+      val (snapshot, blocks, transactions, balances) = mappedGlobalData
```

Comment:
why currency data are in the for comp and global data are in yield? imo only the SplitSnapshot should be in the yield. or even, create a fn in the SplitSnapshot companion object that will receive mapped global data and mapped currency data, and produce the split snapshot. then, that whole for comprehension may look like:
```scala
(globalMapper.mapGlobalSnapshot(...), currencyMapper.mapCurrencySnapshots(...))
  .mapN(SplitSnapshot(_, _))
```